src/eth.rs

```diff
@@ -145,10 +145,14 @@ impl TryFrom<&[u8]> for EIP1559Transaction {
     type Error = ();
 
     fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
-        let [decoded_chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, recipient, value, data, _, _, _]: [Vec<u8>; 11] =
+        let [chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, recipient, value, data, _, _, _]: [Vec<u8>; 11] =
             rlp::decode_list(value).try_into().map_err(|_| ())?;
+        while chain_id.len() < 8 {
+            chain_id.insert(0, 0);
+        }
+        chain_id = u64::from_be_bytes(chain_id.try_into().expect("chain_id must be 8 bytes long after padding"));
```

Comment:
no need to panic here, you can just return an error `...try_into().map_err(|_| ())?`.